### DISCUSSION TITLE
When use pacman to update gdb to a new version, the modified file etc/gdbinit get blindly overwrite by the update, is it by design?

### DISCUSSION THREAD

**Discussion by asmwarrior** [msys2/MINGW-packages#19893] [Q&A]
Hi, sometimes, I need to modify the file `<msys2_root>\mingw64\etc\gdbinit` to include some customized gdb pretty printer, but once I run the `pacman` update command to update the gdb to a new version, this file always get overwriten by the new content. Thus after the update, my changes are lost.

Is this by design?

When discussed with ssbssa at this post, see this comment: https://github.com/ssbssa/gdb/issues/4#issuecomment-1914105681
He said this is not following the way by the pacman's document mentioned here: [pacman/Pacnew and Pacsave](https://wiki.archlinux.org/title/pacman/Pacnew_and_Pacsave)

Any ideas?

Thanks.

**Comment by ssbssa** [comment]
You mean the gcc package, right? Since that's where the gdbinit file comes from.

Also I think gdbinit is just missing the `backup` annotation, as explained [here](https://wiki.archlinux.org/title/pacman/Pacnew_and_Pacsave#Package_backup_files) and [here](https://wiki.archlinux.org/title/PKGBUILD#backup).

**Reply by Biswa96** [reply]
As $MINGW_PREFIX/etc/gdbinit file is created after gcc install, I think the backup option may make things a bit complex. User has to manage that gdbinit file manually.

If the user creates gdbinit file in home directory, pacman would not modify it. See MehdiChinoune's comment below.

**Reply by ssbssa** [reply]
What do you mean it's created after gcc install?
I see the gdbinit file in mingw-w64-x86_64-gcc-13.2.0-3-any.pkg.tar.zst, so `backup=($MINGW_PREFIX/etc/gdbinit)` should *just work* in my opinion.

**Reply by Biswa96** [reply]
Yeah, it should work but user has to manually modify the file after gcc updates.

So, I was saying that let pacman do its thing with /etc/gdbinit and users store their files in ~/.gdbinit. Similar as /etc/nanorc vs ~/.nanorc.

**Reply by ssbssa** [reply]
Yes, but they will be notified when that happens, and can do something about it.
Now it's just overwritten without them knowing.

I don't see a downside to adding the `backup` annotation.

**Comment by asmwarrior** [comment]
This file is very tricky, if you look at the content of this file, it contains such things:

~~~~
python
import sys
sys.path.insert(0, sys.path[0] + '/../../gcc-13.2.0/python')
from libstdcxx.v6.printers import register_libstdcxx_printers
register_libstdcxx_printers (None)
end
~~~~

Note that the gcc version is hard-coded in this file.

I'm not sure it is belong to gcc package or gdb package.

Anyway, can some administrator translate this discussion to an issue? Or do I need to write another issue?

**Comment by asmwarrior** [comment]
Is it possible to remove the hard-coded gcc version strings?

The interesting thing is: I have many folders, such as `mingw64\share\gcc-13.2.0`, `mingw64\share\gcc-xx.x.x`...

So, once a new gcc package is installed, I have a new folder named `gcc-xx.x.x` inside the share folder, so I have to remove those old folders for old gcc versions manually.

**Reply by Biswa96** [reply]
> Is it possible to remove the hard-coded gcc version strings?

The gcc version string will automatically change after gcc version change. See the mingw-w64-gcc/PKGBUILD file.

**Comment by MehdiChinoune** [comment]
You could write your own customized gdbinit and save as $HOME/.gdbinit or $HOME/.config/gdb/gdbinit, It won't be overwritten by any update
https://sourceware.org/gdb/current/onlinedocs/gdb.html/gdbinit-man.html

On native Windows $HOME generally refers to $USERPROFILE
